Sodium/BytesRepresentables/Array+Extension.swift

```diff
@@ -0,0 +1,7 @@
+extension Array: BytesRepresentable where Element == UInt8 {
+    public var bytes: [UInt8] { return self }
+
```

Comment:
Gah, this might be a non-starter – I forgot already that Swift 4.0 doesn't support conditional conformance:
```
error: extension of type 'Array' with constraints cannot have an inheritance clause
extension Array: BytesRepresentable where Element == UInt8 {
^                ~~~~~~~~~~~~~~~~~~
```

Reply:
This could potentially be solved by making `Bytes` a stand-alone type (I.e. a struct that contains a [UInt8] so that `Bytes` can conform to `BytesRepresentable`. The conformances that require a `where` condition can be conditionally enabled in Swift >= 4.1, and Swift 4.0 can still use a constructor to easily create `Bytes` from these specialised generic types. 

Problematically if `Bytes` is a stand-alone type, then we need to further convert to `[UInt8]` before use with the C functions (because these require a `[UInt8]` for the pointer coercion), so perhaps `BytesContainer` would be a better name instead of `Bytes`.

I'm not a major fan of this solution. Anyone got any other ideas here? :)